test/helpers/application_helper_test.rb

```diff
@@ -114,4 +114,18 @@ def test_generate_link_for
         '<a class="btn" href="javascript:foo()">link</a>')
     end
   end
+
+  describe 'host details ui path generation' do
+    setup do
+      Setting[:host_details_ui] = true
+    end
+
+    test 'current_host_details_path should fallback to host id when name is removed during edit' do
+      host = FactoryBot.create(:host, :name => 'test-host')
+      host.name = ''
```

Comment:
Doesn't this work?
```suggestion
      host = FactoryBot.create(:host, :name => '')
```

Reply:
for this `host = FactoryBot.create(:host, :name => '')` does not work as it tries to create a host without name but the validation ` validates :name, :presence => true` in `host/base.rb` kicks in and factory creates the host with a valid name(in my case its creating host with `host1`  name)

so adding `name: ''` in create does not work

Reply:
https://stackoverflow.com/questions/9322353/factory-girl-create-that-bypasses-my-model-validation describes some strategies. This is the quickest now:
```suggestion
      host = FactoryBot.build(:host, :name => '')
      host.save(validate: false)
```

A more structural solution is to implement a trait on the factory itself:
```ruby
trait :without_validations do
  to_create { |instance| instance.save(validate: false) }
end
```

Then it becomes:
```suggestion
      host = FactoryBot.create(:host, :without_validations, :name => '')
```

I like the latter solution because it's reusable and the call also makes it explicit that we're bypassing validations.

Reply:
i have added another test for normal case where host name is present and i kept the create part in setup  which makes host object reusable for both the tests, it would be great if you can check that once